File: common/src/main/java/com/hartwig/actin/treatment/input/single/OneGeneManyCodons.java
package com.hartwig.actin.treatment.input.single;

import java.util.List;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class })
public abstract class OneGeneManyCodons {

    @NotNull
    public abstract String geneName();

    @NotNull
    public abstract List<String> codons();
}

File: common/src/main/java/com/hartwig/actin/treatment/input/single/OneGeneManyCodons.kt
package com.hartwig.actin.treatment.input.single

import org.immutables.value.Value
import org.jetbrains.annotations.NotNull

@Value.Immutable
@Value.Style(passAnnotations = [NotNull::class])
abstract class OneGeneManyCodons() {
    abstract fun geneName(): String
    abstract fun codons(): List<String?>
}
